internal/provider/data_source.go

```diff
@@ -5,77 +5,93 @@ import (
 	"fmt"
 	"math/rand"
 
-	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
-	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
 )
 
-func nullDataSource() *schema.Resource {
-	return &schema.Resource{
+var (
+	_ datasource.DataSource = (*nullDataSource)(nil)
+)
+
+func NewNullDataSource() datasource.DataSource {
+	return &nullDataSource{}
+}
+
+type nullDataSource struct{}
+
+func (n nullDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
```

Comment:
Nit: For consistency, should the `nullDataSource` functions be on a pointer receiver as `NewNullDataSource` returns a pointer?